### ISSUE TITLE
Bad error report when using uninitialized variable in a local function

### ISSUE THREAD

**Issue by ForNeVeR** [dotnet/roslyn#76586]
**Version Used**: 
```
dotnet --info
.NET SDK:
 Version:           9.0.101
 Commit:            eedb237549
 Workload version:  9.0.100-manifests.3068a692
 MSBuild version:   17.12.12+1cce77968
```

**Steps to Reproduce**:

Write the following program.

```csharp
namespace ClassLibrary2;

public class Class1
{
    public static void Foo()
    {
        var my = DoSmth();

        var guid = Guid.NewGuid();
        int DoSmth() => guid.GetHashCode();
    }
}
```

**Diagnostic Id**: 

```
CS0170: Use of possibly unassigned field '_dummyPrimitive'
```

**Expected Behavior**:
Roslyn should report that it's impossible to use `DoSmth()` local function before initializing `guid` local variable.

**Actual Behavior**:
```
Class1.cs(7,18): error CS0170: Use of possibly unassigned field '_dummyPrimitive'
```

So, not only it doesn't reference the actual variable that's not initialized, it also mentions `_dummyPrimitive`, a name of the synthetic field in the corresponding reference assembly and not a real field from `System.Guid` (I assume).

Imagine what will happen if you are working in a function with a dozen of variables, and Roslyn tells you that _some_ of them are not initialized, using an unfamiliar field name when telling that. The error is quite confusing.

**Comment by jhinder** [comment]
Probably a duplicate of #67212.

**Comment by ForNeVeR** [comment]
I saw that issue but not sure honestly.

#67212 seems to describe a more complex situation involving pattern matching and `async`, while this one is much simpler.

There's an identical example in the last comment, though. But I am not sure if all the same arguments apply to this particular case.

So, whether you wish to close this as a duplicate is up to you, folks.

**Comment by jaredpar** [comment]
It's effectively a duplicate. The issue is what error is reported for use of the unassigned local.
